src/Grenade/Core/Shape/Accelerate.hs

```diff
@@ -0,0 +1,25 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE KindSignatures        #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE TypeOperators         #-}
+{-# LANGUAGE StandaloneDeriving    #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE RankNTypes            #-}
+{-|
+Module      : Grenade.Core.Shape
+Description : Dependently typed shapes of data which are passed between layers of a network
+Copyright   : (c) Huw Campbell, 2016-2017
+License     : BSD2
+Stability   : experimental
+
+
+-}
+module Grenade.Core.Shape.Accelerate
+  ( S
+  ) where
+
+import           Data.Array.Accelerate
+
+type S sh = Array sh Double
```

Comment:
I don't think this can work.

The shapes in accelerate have to be formed with `Z` and `:.` (or `DIM1`, `DIM2`, `DIM3`), while it looks like you're using grenade's shapes which hold their dependent type information.